AudiusProject/audius-protocol
# [CON-131] Refactor state machine redis metrics to reduce calls

### Description
- Change fetching sync metrics via 1 redis call per nodeUser to instead fetching sync metrics for all users at once
- Fix bug that caused sync outcomes to be read improperly, which caused the sync success rate to always be 100%
- Add tests for redis batching + making sure that bug doesn't come back. Increased Content Node coverage:
    - line coverage 64.00% -> 66.85%
    - branch coverage 50.85% -> 52.78%

### Tests
The latest commit has integration tests, and if you go back to the first commit there are logs that help with manually verifying functionality -- it performs the original redis check and throws an error with the message `[theo] ERROR MISMATCHED METRICS` if the original metric doesn't match the metric pulled from the new unified redis call.

Steps I followed to test:
1. Use the first commit in this PR (it has extra logs for testing) locally and then on staging
2. Hit the endpoint <node url>/theo/enqueue
3. Search the Docker logs to make sure the state machine operation completed and look for any logs with `[theo]` in them:
    - `docker container ps | grep "cn"`
    - `docker container logs <container for the CN you're looking for> |& less`
    - Type `[theo]` <Enter> (there shouldn't be any hits)
    - Type `processStateMachineOperation` <Enter> (the decision tree should be complete with no failures)

### How will this change be monitored? Are there sufficient logs?
Search for "processStateMachineOperation()" in logs to monitor the decision tree for new errors. `userSecondarySyncMetricsMapLength` should equal `nodeUsersLength`
